BZ1960541: annotations (#39612)
openshift/openshift-docs: modules/nw-route-specific-annotations.adoc

|`haproxy.router.openshift.io/rate-limit-connections.rate-tcp`| Limits the rate at which a client with the same source IP address can make TCP connections. It accepts a numeric value.  +
Note: Using this annotation provides basic protection against distributed denial-of-service (DDoS) attacks. |
|`haproxy.router.openshift.io/timeout` | Sets a server-side timeout for the route. (TimeUnits) | `ROUTER_DEFAULT_SERVER_TIMEOUT`

> |`haproxy.router.openshift.io/timeout-tunnel` | This timeout is effective in the case of tunnel connection, for example, WebSocket over cleartext, edge, reencrypt, or passthrough routes. With cleartext, edge, or reencrypt route types, this annotation is applied as a timeout tunnel with the existing timeout value. For the passthrough route types, the annotation takes precedence over any existing timeout value set. | `ROUTER_DEFAULT_TUNNEL_TIMEOUT`

Reviewer: ```suggestion
|`haproxy.router.openshift.io/timeout-tunnel` | This timeout applies to a tunnel connection, for example, WebSocket over cleartext, edge, reencrypt, or passthrough routes. With cleartext, edge, or reencrypt route types, this annotation is applied as a timeout tunnel with the existing timeout value. For the passthrough route types, the annotation takes precedence over any existing timeout value set. | `ROUTER_DEFAULT_TUNNEL_TIMEOUT`
```
I am not sure if my suggest changes the technical accuracy of the sentence, but I did stumble over `effective`.

> |`ingresses.config/cluster ingress.operator.openshift.io/hard-stop-after` | You can set either an ingresscontroller or the ingress config . Redeploys the router and configures the HA proxy to emit the haproxy `hard-stop-after` global option, which defines the maximum time allowed to perform a clean soft-stop. | `ROUTER_HARD_STOP_AFTER`

Reviewer: ```suggestion
|`ingresses.config/cluster ingress.operator.openshift.io/hard-stop-after` | You can set either an ingresscontroller or the ingress config . This annotation redeploys the router and configures the HA proxy to emit the haproxy `hard-stop-after` global option, which defines the maximum time allowed to perform a clean soft-stop. | `ROUTER_HARD_STOP_AFTER`
```

Reviewer 2: One other comment, should ingresscontroller be in backticks? For example `ingresscontroller`? Just something to think about and review. If not, then you can ignore this comment :)

Author: Based on context I see instances of backticks being used for example IngressController resource or object or API. Based on the context I would not add backticks but I realized the spelling is wrong >> IngressController (correct one) . :)
